src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -576,7 +577,7 @@ static void* VirtualReserveInner(size_t size, size_t alignment, uint32_t flags,
         }
 
         pRetVal = pAlignedRetVal;
-#ifdef MADV_DONTDUMP
+#if defined(MADV_DONTDUMP) && !defined(TARGET_BROWSER)
```

Comment:
Do you believe that the rest of this implementation is going to work on wasm?

runtimelab branch reimplemented it from scratch for wasm: https://github.com/dotnet/runtimelab/blob/feature/NativeAOT-LLVM/src/coreclr/gc/wasm/gcenv.wasm.cpp . https://github.com/dotnet/runtimelab/pull/1510 describes the motivation. Should we adopt this implementation instead of trying to ifdef what's here today?

Reply:
It works so far. The NativeAOT one is simple and more performant, so I will try to use it. I would like to do it in separate PR if possible.